### PR TITLE
Memory/reshape op

### PR DESCRIPTION
**Pull request by dzhwinter**
remove reshape op inplace attribute.

### CODE REVIEW

File: paddle/fluid/operators/reshape_op.cc [hunk 1]
```diff
@@ -251,19 +238,12 @@ class ReshapeGradKernel {
...
-    if (!inplace) {
-      framework::TensorCopy(*d_out, ctx.GetPlace(), ctx.device_context(), d_x);
-      ctx.device_context().Wait();
-      d_x->Resize(in_dims);
-    } else {
-      d_x->ShareDataWith(*d_out);
-      d_x->Resize(in_dims);
-    }
+    framework::TensorCopy(*d_out, ctx.GetPlace(), ctx.device_context(), d_x);
+    ctx.device_context().Wait();
```
Comment (reyoung) [inline paddle/fluid/operators/reshape_op.cc]: TensorCopySync

**Review by reyoung — DISMISSED** [review]
LGTM, However, it will slow down the reshape_op. I will test it in my feature branch.

File: paddle/fluid/operators/reshape_op.cc (hunk 1 above)
Comment (dzhwinter) [reply paddle/fluid/operators/reshape_op.cc]: fix done.

**Review by gongweibao — APPROVED** [review]
LGTM

**Comment by chengduoZH** [comment]
This PR makes the training of Transformer slow. I made some examples on P40 to compare the performance between before and after of this PR, and found that the training speed down to 2.23 step/sec from 2.41 step/sec.

**Pull request merged**